Sources/MuxUploadSDK/Public API/MuxUpload+Photos.swift

```diff
@@ -0,0 +1,58 @@
+//
+//  MuxUpload+PhotosKit.swift
+//
+
+import Foundation
+import Photos
+
+extension MuxUpload {
+    /// Asynchronously creates a MuxUpload using a ``PHAsset``
+    ///
+    /// - Parameters:
+    ///    - imageManager: Photos image manager
+    ///    - inputAsset: the upload input asset, its
+    ///    ``PHAsset.mediaType`` needs to be ``PHAssetMediaType.video``
+    ///    - uploadURL: the direct upload URL
+    ///    - transportSettings: upload transport settings
+    ///    - optOutOfEventTracking: opts out of SDK event reporting
+    ///    - standardizationSettings: if enabled the SDK checks
+    ///    if the input file is in a standard input format,
+    ///    attempts local conversion if it is non-standard
+    ///    - completion: receives the initialized MuxUpload
+    ///    when it is ready, receives nil if initialization
+    ///    failed
+    public static func makeByExporting(
```

Comment:
An `init` accepting a `PHAsset` parameter adds requirements for additional state management to `MuxUpload`. The additional state handling needed because `PHAsset` -> `AVAsset` conversion is an async operation.

We can alternatively expose this API via an extension on `PHAsset` or `PHImageManager`. Any thoughts or opinions?

(Let's punt on `AVAssetExportSession`-related discussion here, the next follow-on PR for input standardization will thread `AVAssetExportSession` throughout the SDK so can revisit anything AVFoundation-related (including what was introduced in this first PR, there.)

Reply:
That said `makeByExporting` spelling totally up for debate here, am not too keen on it to be honest.

Reply:
How would you feel about 
```
extension PHAsset {
  func prepareForUpload(from: imageManager, ..., (MuxUpload?) -> Void)
```
or `prepareForUploadToMux(...)`

Reply:
I like `prepareForUpload` the most of these options.

Reply:
Also, just pointing out this specific method only takes over part of the export flow. We may want to include another convenience method to cover the call to `PHAsset.fetchAssets`:

```
extension PHPickerResult {
  /// Fetches the asset from this Picker Result and asynchronously prepares it for upload
  func prepareForUpload(from: PHImageManager, ..., completion: (MuxUpload?) -> Void)
```

Reply:
I do like the `prepareForUpload` but to add some specificity there what about `prepareForDirectUpload`?

Reply:
> Also, just pointing out this specific method only takes over part of the export flow. We may want to include another convenience method to cover the call to `PHAsset.fetchAssets` (which is how the `PHAsset` is created):
> 
> ```
> extension PHPickerResult {
>   /// Fetches the asset from this Picker Result and asynchronously prepares it for upload
>   func prepareForUpload(from: PHImageManager, ..., completion: (MuxUpload?) -> Void)
> ```

Yes, taking over only a part of the export flow was intentional. 

Two considerations:

- Balancing how much the API handles on behalf of the client vs how many knobs it exposes (the more it handles, the more complex the interactions with client logic)
- Preserving API un-opinionatedness with respect to operations outside the core functionality of the SDK

We can definitely iterate on the shape of this API and there may well be room for supplementary API(s) that do more for relatively standard use cases.

Reply:
Resolving this, feel free to comment again if anything outstanding remains